### PR TITLE
Remove remaining NETWARE ifdef's

### PR DESCRIPTION
**Pull request by richsalz**
The config stuff was partially removed in 1.1.0; this removes the last vestiges.

### CODE REVIEW

**Comment by richsalz** [comment]
Updated CHANGES (not NEWS), please reconfirm.

File: include/internal/sockets.h [hunk 1]
```diff
@@ -116,14 +116,11 @@ struct servent *PASCAL getservbyname(const char *, const char *);
...
+ * Some IPv6 implementations are broken, you can disable them in known
+ * bad versions.
  */
 # if !defined(OPENSSL_USE_IPV6)
-#  if defined(AF_INET6) && !defined(NETWARE_CLIB)
-#   define OPENSSL_USE_IPV6 1
-#  else
-#   define OPENSSL_USE_IPV6 0
-#  endif
+#  define OPENSSL_USE_IPV6 1
```
Comment (vdukhovni) [inline include/internal/sockets.h]: Presumably we still need `if defined(AF_INET6)`

Comment (richsalz) [reply include/internal/sockets.h]: No, it's only needed if you don't define use_ipv6 which our config normally does now.

Comment (vdukhovni) [reply include/internal/sockets.h]: I don't have the cycles to evaluate this answer. Just noticing that in this one case the patch was changing things not only for NETWARE, but rather also unconditionally enabling IPv6 even when AF_INET6 did not appear to be defined. Perhaps that's OK...

Comment (levitte) [reply include/internal/sockets.h]: Actually, @vdukhovni is correct.  @richsalz, *this* *is* the spot where `OPENSSL_USE_IPv6` gets defined for the rest of the source, nowhere else.  The reason for these kinda convoluted conditions is to give the builder the option to disable IPv6 support like like this:

```
./config -DOPENSSL_USE_IPV6=0
```

So the above should still read like this:

```
# if !defined(OPENSSL_USE_IPV6)
#  if defined(AF_INET6)
#   define OPENSSL_USE_IPV6 1
#  else
#   define OPENSSL_USE_IPV6 0
#  endif
# endif

Comment (richsalz) [reply include/internal/sockets.h]: Thanks!  See https://github.com/openssl/openssl/pull/5030

**Comment by vdukhovni** [comment]
Not sure why #5030 is a separate PR, rather than an update of this one, but either way the combined change is OK, so feel free to combine or push as separate commits if that's for some reason preferable.

**Comment by levitte** [comment]
@vdukhovni, it seems that @richsalz merged earlier but forgot to close this PR...

**Comment by vdukhovni** [comment]
Moot. Already merged.

**Pull request closed without merging**